Core/src/main/java/org/tribuo/impl/ArrayExample.java

```diff
@@ -159,12 +159,19 @@ public ArrayExample(T output, List<? extends Feature> features) {
      */
     public ArrayExample(Example<T> other) {
         super(other);
-        featureNames = new String[other.size()];
-        featureValues = new double[other.size()];
-        for (Feature f : other) {
-            featureNames[size] = f.getName();
-            featureValues[size] = f.getValue();
-            size++;
+        if(other instanceof ArrayExample) {
+            ArrayExample< T> otherArr = (ArrayExample<T>) other;
+            featureNames = Arrays.copyOf(otherArr.featureNames, otherArr.featureNames.length);
```

Comment:
I know I wrote this, but the Arrays.copyOf should probably be to `size` not `otherArr.featureNames.length` to make it match up with the other side of the else. Plus it's slightly more memory efficient that way.

Reply:
I presume you mean otherArr.size here.

Reply:
Yeah, that one.